### PR TITLE
[JENKINS-68591] Use new layout for remote class loader stats

### PR DESCRIPTION
**Pull request by MarkEWaite**
## [JENKINS-68591](https://issues.jenkins.io/browse/JENKINS-68591) Use new layout for remote class loader statistics

The groovy generated layout matches the jelly layout.  If others have a better approach to improve it, I am open to suggestions.

No tests are included in the pull request because it is difficult to assert table layout from an automated test.

### Before

current-remote-class-loader-statistics

### After

new-remote-class-loader-statistics

## Concern

The table expands to full width of the page.  Table readability is reasonable thanks to the change offered by @NotMyFault to use the left alignment rather than right alignment of the numerical data in the table

### Proposed changelog entries

* Show remote class loader statistics of agents with new table layout.

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [x] (If applicable) Jira issue is well described
- [x] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change) and are in the imperative mood. [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [x] Appropriate autotests or explanation to why this change has no tests
- [x] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadoc, as appropriate. 
- [x] For dependency updates: links to external changelogs and, if possible, full diffs

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [x] There are at least 2 approvals for the pull request and no outstanding requests for change
- [x] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [x] Changelog entries in the PR title and/or `Proposed changelog entries` are accurate, human-readable, and in the imperative mood
- [x] Proper changelog labels are set so that the changelog can be generated automatically
- [x] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [x] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).

### CODE REVIEW

**Review by NotMyFault — COMMENTED** [review]
Looks nice so far, but I'd recommend to wrap the table header into the thead tag, to align it with the design used on other pages:

Bildschirmfoto 2022-05-20 um 15 48 57

https://gist.github.com/NotMyFault/335dc50bfe7e7501510e50af7a182199 to what I mean above

**Review by NotMyFault — APPROVED** [review]

**Review by timja — APPROVED** [review]

File: core/src/main/resources/jenkins/slaves/systemInfo/ClassLoaderStatisticsSlaveInfo/systemInfo.groovy [hunk 1]
```diff
@@ -1,20 +1,21 @@
...
-def right = 'text-align: right'
 if (my instanceof SlaveComputer) {
     SlaveComputer c = my
 
-    table(class: 'bigtable') {
-        tr {
-            th _('Loading Type')
-            th _('Time (s)')
-            th _('Count')
+    table(class: 'jenkins-table') {
```
Comment (mawinter69) [inline core/src/main/resources/jenkins/slaves/systemInfo/ClassLoaderStatisticsSlaveInfo/systemInfo.groovy:7]: `width: auto` should make the table compact

Comment (MarkEWaite) [reply core/src/main/resources/jenkins/slaves/systemInfo/ClassLoaderStatisticsSlaveInfo/systemInfo.groovy:7]: I wasn't able to find a way to apply `width: auto` in that change and have any different result than is shown in the screenshot.

I also tried [`table-layout: auto`](https://www.w3schools.com/cssref/pr_tab_table-layout.asp) without success

Comment (mawinter69) [reply core/src/main/resources/jenkins/slaves/systemInfo/ClassLoaderStatisticsSlaveInfo/systemInfo.groovy:7]: In my plugin I do
`<table class="jenkins-table sortable" style="width: auto">`
which works fine.
In the old styles there was `bigtable--compact` seems this is missing

Comment (MarkEWaite) [reply core/src/main/resources/jenkins/slaves/systemInfo/ClassLoaderStatisticsSlaveInfo/systemInfo.groovy:7]: Thanks for the pointer.  Much appreciated.

I was able to make it work with `style: "width: auto"` but I prefer the layout without the `width: auto`.  The `width: auto` makes the separation between column headings less clear to me.  Open to other opinions, but I prefer it full width.  I'll leave it with the current layout unless there is strong feedback that it needs more styling.

Comment (NotMyFault) [reply core/src/main/resources/jenkins/slaves/systemInfo/ClassLoaderStatisticsSlaveInfo/systemInfo.groovy:7]: I'd rather not do that, as that breaks with the styling of existing tables.

Comment (MarkEWaite) [reply core/src/main/resources/jenkins/slaves/systemInfo/ClassLoaderStatisticsSlaveInfo/systemInfo.groovy:7]: > I'd rather not do that, as that breaks with the styling of existing tables.

When you say, "I'd rather not do that", I assume you mean "I would rather we not add `width: auto` to the table".  Did I understand correctly @NotMyFault ?

**Comment by basil** [comment]
> In the old styles there was `bigtable--compact` seems this is missing

Please file a UX regression ticket to cover this loss of functionality in the CSS API explaining the reason this functionality is desired. If the removal of this functionality was done for stylistic reasons that supersede the reason given in the ticket, the ticket can be closed as "not a defect." If the removal of this functionality was not done for stylistic reasons and the reason given in the ticket is valid, the ticket can remain open.

**Comment by MarkEWaite** [comment]
This PR is now ready for merge.  We will merge it after approximately 24 hours if there is no negative feedback.

File: core/src/main/resources/jenkins/slaves/systemInfo/ClassLoaderStatisticsSlaveInfo/systemInfo.groovy [hunk 2]
```diff
@@ -1,20 +1,21 @@
...
+        thead {
+            tr {
+                th { text(_('Loading Type')) }
+                th { text(_('Time (s)')) }
+                th { text(_('Count')) }
```
Comment (daniel-beck) [inline core/src/main/resources/jenkins/slaves/systemInfo/ClassLoaderStatisticsSlaveInfo/systemInfo.groovy:12]: Why this change?

Comment (timja) [reply core/src/main/resources/jenkins/slaves/systemInfo/ClassLoaderStatisticsSlaveInfo/systemInfo.groovy:12]: specifically? count is there before

Comment (daniel-beck) [reply core/src/main/resources/jenkins/slaves/systemInfo/ClassLoaderStatisticsSlaveInfo/systemInfo.groovy:12]: The change from `th _('key')` to `th { text(_('key')) }`

Comment (MarkEWaite) [reply core/src/main/resources/jenkins/slaves/systemInfo/ClassLoaderStatisticsSlaveInfo/systemInfo.groovy:12]: I made that change because I didn't find a way to make it render correctly with one of the iterations that was using

```
th(style: right) _('Time (s)')
```

When that wouldn't work for me (non-existent Javascript / CSS skills), I switched to something that worked for me:

```
th(style: right) { text(_('Time (s)')) }
```

That was based on entries elsewhere in the file.

I'm more than happy to restore it to the earlier form, especially if there is a benefit from that change:

```
th _('Time (s)')
```

Is there a benefit from returning it to the original text?
